lib/mongo/crypt/hooks.rb

```diff
@@ -88,6 +88,19 @@ def hash_sha256(input)
         Digest::SHA2.new(256).digest(input)
       end
       module_function :hash_sha256
+
+      # An RSASSA-PKCS1-v1_5 with SHA-256 signature function.
+      #
+      # @param [ String ] key The PKCS#8 encrypted private key.
+      # @param [ String ] input The data to be signed.
+      #
+      # @return [ String ] The signature.
+      def rsaes_pkcs_signature(key, input)
```

Comment:
Where are the requirements/specification for this method?

Reply:
Some information is available here - https://docs.google.com/document/d/18rO1QYORNvKs8rYWtXKdrT-2YaORMJ-r8xcM8g1SJ8Q/edit#heading=h.h2339z2vnbng

There are also some further links:
https://www.rfc-editor.org/rfc/rfc7518.html#section-3.3
https://www.rfc-editor.org/rfc/rfc3447#section-8.2

Reply:
I think it would be helpful to add a unit test that signs the "data to sign" as per that section in the google doc. The implementation looks good to me. The other part that that test would possibly clarify is whether the signature as produced by ruby's openssl binding is in the expected format vs mongocrypt/that doc (e.g. base64 encoded?)

Reply:
Test added.